demos/vc_issuer/src/main.rs

```diff
@@ -0,0 +1,521 @@
+
+    SIGNATURES.with(|sigs| {
+        let mut sigs = sigs.borrow_mut();
+        add_signature(&mut sigs, msg_hash, seed);
+    });
+    update_root_hash();
+    PrepareCredentialResponse::Ok(PreparedCredentialData {
+        prepared_context: Some(ByteBuf::from(credential_jwt.as_bytes())),
+    })
+}
+
+fn update_root_hash() {
+    use ic_certified_map::labeled_hash;
+    SIGNATURES.with(|sigs| {
+        let sigs = sigs.borrow();
+        let prefixed_root_hash = &labeled_hash(LABEL_SIG, &sigs.root_hash());
+        set_certified_data(&prefixed_root_hash[..]);
+    })
+}
+
+#[query]
+#[candid_method(query)]
+fn get_credential(req: GetCredentialRequest) -> GetCredentialResponse {
+    if let Err(e) = authorize_vc_request(&req.signed_id_alias) {
+        return GetCredentialResponse::Err(e);
+    };
+    if let Err(err) = verify_credential_spec(&req.credential_spec) {
+        return GetCredentialResponse::Err(IssueCredentialError::UnsupportedCredentialSpec(err));
+    }
+    let subject_principal = req.signed_id_alias.id_alias;
+    let seed = calculate_seed(&subject_principal);
+    let canister_id = ic_cdk::id();
+    let canister_sig_pk = CanisterSigPublicKey::new(canister_id, seed.to_vec());
+    let prepared_context = match req.prepared_context {
+        Some(context) => context,
+        None => return GetCredentialResponse::Err(internal_error("missing prepared_context")),
+    };
+    let credential_jwt = match String::from_utf8(prepared_context.into_vec()) {
+        Ok(s) => s,
+        Err(_) => return GetCredentialResponse::Err(internal_error("invalid prepared_context")),
+    };
+    let signing_input =
+        vc_signing_input(&credential_jwt, &canister_sig_pk).expect("failed getting signing_input");
+    let msg_hash = vc_signing_input_hash(&signing_input);
+    let maybe_sig = SIGNATURES.with(|sigs| {
+        let sigs = sigs.borrow();
+        get_signature(&sigs, seed, msg_hash)
+    });
+    let sig = if let Some(sig) = maybe_sig {
+        sig
+    } else {
+        return GetCredentialResponse::Err(IssueCredentialError::SignatureNotFound(String::from(
+            "signature not prepared or expired",
+        )));
+    };
+    let vc_jws =
+        vc_jwt_to_jws(&credential_jwt, &canister_sig_pk, &sig).expect("failed constructing JWS");
+    GetCredentialResponse::Ok(IssuedCredentialData { vc_jws })
+}
+
+#[update]
+#[candid_method]
+async fn vc_consent_message(req: Icrc21VcConsentMessageRequest) -> Icrc21ConsentMessageResponse {
+    if let Err(err) = verify_credential_spec(&req.credential_spec) {
+        return Icrc21ConsentMessageResponse::Err(Icrc21Error::NotSupported(Icrc21ErrorInfo {
+            description: err,
+            error_code: 0,
+        }));
+    }
+    Icrc21ConsentMessageResponse::Ok(Icrc21ConsentInfo {
+        consent_message: get_vc_consent_message(&req),
+        language: "en-US".to_string(),
+    })
+}
+
+fn verify_credential_spec(spec: &CredentialSpec) -> Result<(), String> {
```

Comment:
Thinking of how this might be helpful for testing:

How about having `add_employee` take both a principal _and_ an `employerName`? Then no employers have to be hard coded

Reply:
Agreed, this could be useful in the future.  Let's add it once we need it.